### ISSUE TITLE
create database error

### ISSUE THREAD

**Issue by yanwei49** [tekartik/sqflite#978]
///   var databasesPath = await getDatabasesPath();
  ///   String path = join(databasesPath, 'demo.db');
  ///   Database database = await openDatabase(path, version: 1,
  ///       onCreate: (Database db, int version) async {
  ///     // When creating the db, create the table
  ///     await db.execute(
  ///         'CREATE TABLE Test (id INTEGER PRIMARY KEY, name TEXT, value INTEGER, num REAL)');
  ///   });
I will report an error when using the above method to create a clock in database.

databaseOpenHelper.openDatabase()
error:
SqfliteFfiException (SqfliteFfiException(sqlite_error: 14, , open_failed: SqliteException(14): while opening the database, bad parameter or other API misuse, bad parameter or other API misuse (code 21)}) DatabaseException(open_failed: SqliteException(14): while opening the database, bad parameter or other API misuse, bad parameter or other API misuse (code 21)))

**Comment by alextekartik** [comment]
I don't know on which platform you are getting the exception (iOS, Android, Windows, Linux, MacOS?). `getDatabasesPath()` makes sense on Android but maybe not on other platform. Maybe, make sure that the parent folder exists first.

**Comment by azwalzaman** [comment]
Same error on Android

**Comment by alextekartik** [comment]
`getDatabasesPath()` only makes sense on Android using sqflite. Using ffi (I guess that is what you are using @azwalzaman right?), it returns a dummy local path (ok for testing). In you app, you should use `path_provider` to get the relevant location to build the proper path and make sure the parent folder exists before opening a database.

**Comment by azwalzaman** [comment]
> `getDatabasesPath()` only makes sense on Android using sqflite. Using ffi (I guess that is what you are using @azwalzaman right?), it returns a dummy local path (ok for testing). In you app, you should use `path_provider` to get the relevant location to build the proper path and make sure the parent folder exists before opening a database.

You're right. I'm sorry I missed the part in the docs saying the implementation for `getDatabasesPath()` is lame. I was able to use `path_provider` to make it work. Thankyou!
